### PR TITLE
GRIM: Support Chinese Monkey4

### PR DESCRIPTION
**Pull request by phcoder**

### CODE REVIEW

File: engines/grim/font.cpp [hunk 1]
```diff
@@ -175,7 +177,55 @@ void BitmapFont::load(const Common::String &filename, Common::SeekableReadStream
...
+	_lastChar = 0xffff;
+
+	_isDBCS = false;
+	_isUnicode = true;
+
+	// Read character headers
+	_charHeaders = new CharHeader[_numChars];
+
+	// Read character indexes - are the key/value reversed?
+	_fwdCharIndex.resize(0x10000, -1);
```
Comment (aquadran) [inline engines/grim/font.cpp:204]: what 0x10000 magic number do? add macro?

File: engines/grim/font.cpp [hunk 2]
```diff
@@ -175,7 +177,55 @@ void BitmapFont::load(const Common::String &filename, Common::SeekableReadStream
...
+  	Image::TGADecoder dec;
+	bool success = dec.loadStream(*image);
+
+	if (!success)
+		return;
+
+	const Graphics::Surface *surf = dec.getSurface();
+
+	_filename = filename;
+	_numChars = index->size() / 10;
```
Comment (aquadran) [inline engines/grim/font.cpp:190]: macro for 10?

File: engines/grim/font.cpp (hunk 1 above)
Comment (phcoder) [reply engines/grim/font.cpp]: It's highest uint16 + 1. Added a macro

File: engines/grim/font.cpp (hunk 2 above)
Comment (phcoder) [reply engines/grim/font.cpp]: It's a size of index entry. Added a const

**Pull request merged**